module/applications/item/ability-use-dialog.mjs

```diff
@@ -410,4 +426,54 @@ export default class AbilityUseDialog extends Dialog {
     const value = uses.value - consume;
     return value <= 0;
   }
+
+  /* -------------------------------------------- */
+  /*  Event Handlers                              */
+  /* -------------------------------------------- */
+
+  /** @inheritDoc */
+  activateListeners(jQuery) {
+    super.activateListeners(jQuery);
+    const [html] = jQuery;
+
+    html.querySelector('[name="slotLevel"]')?.addEventListener("change", this._onChangeSlotLevel.bind(this));
+  }
+
+  /* -------------------------------------------- */
+
+  /**
+   * Update summoning profiles when spell slot level is changed.
+   * @param {Event} event  Triggering change event.
+   */
+  _onChangeSlotLevel(event) {
+    const level = parseInt(event.target.value.replace("spell", ""));
+    const item = this.item.clone({ "system.level": level });
+    const summoningData = this.constructor._createSummoningOptions(item);
+    const originalInput = this.element[0].querySelector('[name="summonsProfile"]');
+    if ( !originalInput ) return;
+
+    // If only one profile, replace with hidden input
+    if ( !summoningData.profiles ) {
```

Comment:
Prefer to reverse these branches so that the `else` condition doesn't read like a double-negative